lib/Token/Token.cpp

```diff
@@ -223,6 +236,19 @@ bool Token::isBasePointerType() const {
   return false;
 }
 
+bool Token::isValPointerType() const {
+ if (this->Ty->isPointerTy())
+	return true;
+ return false;
+}
+
+bool Token::checkIsNull() const {
+ llvm::errs() << "\n Inside checkIsNull";
+ if (this == nullptr)
```

Comment:
Does this function check if the Token object is nullptr or not? but if it is then how will it call this function?